logging/troubleshooting/cluster-logging-troubleshooting-for-critical-alerts.adoc

```diff
@@ -0,0 +1,538 @@
+
+.. Check if the Elasticsearch JVM Heap usage is high.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c <elasticsearch_pod_name> -- es_util --query=_nodes/stats?pretty
+----
++
+In the command output, review the `node_name.jvm.mem.heap_used_percent` field to determine the JVM Heap usage.
+
+.. Check for high CPU utilization.
+
+.Additional resources
+
+* See "Free up or increase disk space" in the Elasticsearch topic, link:https://www.elastic.co/guide/en/elasticsearch/reference/7.13/fix-common-cluster-issues.html#fix-red-yellow-cluster-status[Fix a red or yellow cluster status].
+
+[id="elasticsearch-cluster-health-is-yellow"]
+== Elasticsearch Cluster Health is Yellow
+
+Replica shards for at least one primary shard are not allocated to nodes.
+
+.Troubleshooting
+
+. Check the disk space of the Elasticsearch node.
+
+. Increase the node count or the disk space of existing nodes by adjusting `nodeCount` or `storage.size` in the `ClusterLogging` CR.
+
+.Additional resources
+
+* See *Elasticsearch Disk Usage* in xref:../../logging/cluster-logging-dashboards.adoc#cluster-logging-dashboards-logging_cluster-logging-dashboards[OpenShift Logging dashboards].
+* xref:../../logging/config/cluster-logging-configuring-cr.adoc#cluster-logging-configuring-crd_cluster-logging-configuring-cr[About the Cluster Logging custom resource]
+* xref:../../logging/config/cluster-logging-log-store.adoc#cluster-logging-elasticsearch-storage_cluster-logging-store[Configuring persistent storage for the log store]
+
+* See "Free up or increase disk space" in the Elasticsearch topic, link:https://www.elastic.co/guide/en/elasticsearch/reference/7.13/fix-common-cluster-issues.html#fix-red-yellow-cluster-status[Fix a red or yellow cluster status].
+
+
+// [id="elasticsearch-write-requests-rejection-jumps"]
+// == Elasticsearch Write Requests Rejection Jumps
+//
+// .Troubleshooting
+// TBD
+// Note for writer: This is a warning alert and we haven't documented troubleshooting steps for warning alerts yet. I guess you can skip this in currrent release.
+
+[id="elasticsearch-node-disk-low-watermark-reached"]
+== Elasticsearch Node Disk Low Watermark Reached
+
+Elasticsearch does not allocate shards to nodes that https://www.elastic.co/guide/en/elasticsearch/reference/6.8/disk-allocator.html[reach the low watermark].
+
+.Troubleshooting
+
+. Identify the node on which Elasticsearch is deployed.
++
+[source,terminal]
+----
+oc -n openshift-logging get po -o wide
+----
+
+. Check if there are `unassigned shards`.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_cluster/health?pretty | grep unassigned_shards
+----
+
+. If there are unassigned shards, check the disk space on each node.
++
+[source,terminal]
+----
+oc exec -n openshift-logging -c elasticsearch <elasticsearch_pod_name> -- es_util --query=_nodes/stats/fs?pretty
```

Comment:
```suggestion
$for pod in `oc get po -l component=elasticsearch -o jsonpath='{.items[*].metadata.name}'`; do echo $pod; oc exec -c elasticsearch $es_pod -- df -h /elasticsearch/persistent; done
```
@ehvs Is this what you're suggesting?

Reply:
That would be

```
$for pod in `oc -n openshift-logging get po -l component=elasticsearch -o jsonpath='{.items[*].metadata.name}'`; do echo $pod; oc -n openshift-logging exec -c elasticsearch $pod -- df -h /elasticsearch/persistent; done
```

the **for** loop, export a variable **pod** and it can be applied in the **oc exec** command. I tweaked a bit further to include the namespace, just in case the user is outside the correct project.